src/test/unit/scheduler/test_compare.py

```diff
@@ -39,14 +43,17 @@ def setUp(self):
         self.config.set('ExpertSettings', 'block_delay', '2')
         self.config.set('ExpertSettings', 'ssdeep_ignore', '80')
 
-        self.bs_patch = unittest.mock.patch(target='storage.binary_service.BinaryService', new=lambda _: MockDbInterface)
-        self.bs_patch.start()
+        self.bs_patch_new = unittest.mock.patch(target='storage.binary_service.BinaryService.__new__', new=lambda *_, **__: MockDbInterface())
+        self.bs_patch_init = unittest.mock.patch(target='storage.binary_service.BinaryService.__init__', new=lambda _: None)
```

Comment:
Do we need both patches or does `__new__` suffice?

Reply:
sadly, we need both because they are called individually (instead of one calling the other)
I'm sure there is a better way to mock classes (something like `patch(target='storage.binary_service.BinaryService', MockBinaryService)` but that does not seem to work) but I have yet to find one.